src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/DataSourceParserTest/DataSourceParserTest.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace Microsoft.Data.SqlClient.ManualTesting.Tests.SQL.DataSourceParserTest
+{
+    public class DataSourceParserTest
+    {
+        [ConditionalTheory(typeof(DataTestUtility), nameof(DataTestUtility.IsNotAzureServer), nameof(DataTestUtility.IsNotAzureSynapse))]
+        [InlineData("localhost")]
+        [InlineData("tcp:localhost")]
+        [InlineData("np:localhost")]
+        [InlineData(" localhost ")]
+        [InlineData(" tcp:localhost ")]
+        [InlineData(" np:localhost ")]
+        [InlineData(" localhost")]
+        [InlineData(" tcp:localhost")]
+        [InlineData(" np:localhost")]
+        [InlineData("localhost ")]
+        [InlineData("tcp:localhost ")]
+        [InlineData("np:localhost ")]
+        public void ParseDataSourceWithoutInstanceNorPortTestShouldSucceed(string dataSource)
+        {
+            DataTestUtility.ParseDataSource(dataSource, out string hostname, out _, out _);
+            Assert.Equal("localhost", hostname);
+        }
+
+        [ConditionalTheory(typeof(DataTestUtility), nameof(DataTestUtility.IsNotAzureServer), nameof(DataTestUtility.IsNotAzureSynapse))]
+        [InlineData("localhost,1433")]
+        [InlineData("tcp:localhost,1433")]
+        [InlineData("np:localhost,1433")]
+        [InlineData(" localhost,1433 ")]
+        [InlineData(" tcp:localhost,1433 ")]
+        [InlineData(" np:localhost,1433 ")]
```

Comment:
NIT: these are for trim test? If yes, one of them should suffice

Reply:
For completeness sake, both specified protocols, tcp and np, were included in all the DataSourceParser tests.

As you pointed out, the parsing of both tcp and np protocols will both result in the same behavior for parsing data source with space and colon in it.  Thus, one of them should suffice. The np protocol data source has been dropped in all unit tests of DataSourceParser.